adapters/dmx/dmx.go

```diff
@@ -77,6 +86,18 @@ func (adapter *DmxAdapter) MakeRequests(request *openrtb.BidRequest, req *adapte
 		appPublisherCopy := *request.App.Publisher
 		dmxReq.App = &appCopy
 		dmxReq.App.Publisher = &appPublisherCopy
+		if dmxReq.App.Publisher.ID == "" {
+			dmxReq.App.Publisher.ID = publisherId
+		}
+		if rootExtInfo.Bidder.MemberId != "" {
+			dmxRawPubId.Dmx.Id = rootExtInfo.Bidder.MemberId
+		} else {
+			dmxRawPubId.Dmx.Id = rootExtInfo.Bidder.PublisherId
+		}
```

Comment:
We could probably use `UserSellerOrPubId` instead of the repetitive if/else statement:
```
74     if len(request.Imp) >= 1 {
75         err := json.Unmarshal(request.Imp[0].Ext, &rootExtInfo)
76         if err != nil {
77             errs = append(errs, err)
78         } else {
79             publisherId = UserSellerOrPubId(rootExtInfo.Bidder.PublisherId, rootExtInfo.Bidder.MemberId)
80             sellerId = rootExtInfo.Bidder.SellerId
81         }
82     }
83
84     if request.App != nil {
85         appCopy := *request.App
86         appPublisherCopy := *request.App.Publisher
87         dmxReq.App = &appCopy
88         dmxReq.App.Publisher = &appPublisherCopy
89         if dmxReq.App.Publisher.ID == "" {
90             dmxReq.App.Publisher.ID = publisherId
91         }
92 -       if rootExtInfo.Bidder.MemberId != "" {
93 -           dmxRawPubId.Dmx.Id = rootExtInfo.Bidder.MemberId
94 -       } else {
95 -           dmxRawPubId.Dmx.Id = rootExtInfo.Bidder.PublisherId
96 -       }
   +       dmxRawPubId.Dmx.Id = UserSellerOrPubId(rootExtInfo.Bidder.MemberId, rootExtInfo.Bidder.PublisherId)
adapters/dmx/dmx.go
```

Reply:
Using the creating function to forward memberid or publisher_id when declare